### PR TITLE
SPOC-489: Add bidirectional large object migration between native and lolor storage

### PR DESCRIPTION
**Pull request by danolivo**
Adds `lolor.migrate_from_native()` and `lolor.migrate_to_native()` SQL functions
that move large objects between PostgreSQL's native `pg_catalog` storage and
lolor's replication-compatible tables, preserving OIDs, owners, ACLs, and data.

### Design decisions

- **Native → lolor migration is manual**: the user calls `migrate_from_native()` explicitly after `CREATE EXTENSION lolor`. This requires superuser and should be done during a maintenance window.
- **Lolor → native migration is automatic**: `DROP EXTENSION lolor` triggers `migrate_to_native()` via the event trigger, ensuring no large objects are lost on extension removal. If migration fails (e.g. OID conflict), the DROP is rejected so the user can resolve the issue and retry.
- **Migration is safe only in master-replica configurations** for now. In multi-master setups, migrated OIDs lack node-encoding and may conflict with lolor-generated OIDs on other nodes.

### Implementation details

- Forward migration uses bulk `INSERT` for performance; reverse migration uses the native LO API (`lo_create_orig`/`lo_lseek64_orig`/`lowrite_orig`) since direct `INSERT` into `pg_catalog.pg_largeobject` is not allowed from SQL.
- `LOBLKSIZE` is derived at runtime from `current_setting('block_size')` to support non-default block sizes.
- Uses `lo_lseek64_orig` with `bigint` offsets to handle large objects exceeding 2 GB.
- The event trigger guards `migrate_to_native()` with a `pg_proc` existence check for backward compatibility with lolor versions < 1.2.3.
- Row count verification after each bulk copy detects silent data loss.
- Materializes the OID list before `lo_unlink_orig` to avoid scanning the catalog while modifying it.
- Emits `NOTICE` during installation if active streaming replicas are detected.

### Tests

- Forward migration: two native LOs migrated, data integrity verified through lolor API.
- Reverse migration via `DROP EXTENSION`: all objects (including ones created directly in lolor) restored to native storage with data verification.
- Manual `migrate_to_native()` without dropping the extension.
- OID conflict detection for both `migrate_from_native` and `migrate_to_native`.
- `DROP EXTENSION` rejected on OID conflict, user resolves and retries successfully.
- Edge cases: empty native storage, empty lolor storage, extension upgrade path 1.2.2 → 1.2.3.
- TAP test for standby promotion with lolor installed.

### CODE REVIEW

File: lolor--1.2.2--1.3.sql [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+  IF EXISTS (
+    SELECT 1
+    FROM lolor.pg_largeobject_metadata lm
+    JOIN pg_catalog.pg_largeobject_metadata native ON native.oid = lm.oid
+  ) THEN
+    RAISE EXCEPTION 'OID conflict: some lolor large objects already exist in native storage';
+  END IF;
+
+  -- Migrate each object using the native LO API (_orig functions).
+  -- We cannot INSERT directly into pg_catalog.pg_largeobject from SQL,
```
Comment (rasifr) [inline lolor--1.2.2--1.3.sql:157]: Superusers can do INSERT directly into pg_catalog.pg_largeobject. That restriction applies only for non-superusers. Since both functions already require superuser, the loop can be replaced with bulk INSERT ... SELECT, the same approach migrate_from_native() uses.

**Review by mason-sharp — APPROVED** [review]
Important feature. I changed the version number to be 1.3.
